debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/Replicator.java

```diff
@@ -296,6 +296,7 @@ protected boolean performInitialSync() {
             Thread.interrupted();
             aborted.set(true);
         }
+        this.copyThreads.shutdown();
```

Comment:
This won't wait for the tasks to complete (see its JavaDoc). A call to `awaitTermination()` is in order.

Reply:
IMHO it will work
1. see the `latch.await()`, thw shutdown is call only after all taskas are completed
2. even if the task is in progress it will send a signal and tasks in progress will be completed
3. we do not need to block until threads are finished, this can be done asynchronously

Reply:
True that, I missed the latch.